### PR TITLE
Add quantityFilled and its update

### PR DESCRIPTION
**Pull request by XaosSintez**
## Summary
This PR adds `quantityFilled` and ability to set it.

## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

File: openapi/components/requestBodies/SubscriptionItemUpdate.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+content:
+  application/json:
+    schema:
+      type: object
+      required:
+        - quantityUsed
+      properties:
+        quantityUsed:
```
Comment (adamaltman) [inline openapi/components/requestBodies/SubscriptionItemUpdate.yaml:8]: I wonder if it should be `quantityUsed` or `usedQuantity`?
```suggestion
        quantityUsed:
          x-badge: Experimental
```
Even though the badge may not display without some tweaks, we should note it.

File: openapi/components/requestBodies/SubscriptionItemUpdate.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+          description: Used quantity of the subscription item.
```
Comment (adamaltman) [inline openapi/components/requestBodies/SubscriptionItemUpdate.yaml:9]:
```suggestion
          description: Used quantity of the subscription item (experimental property).
```

File: openapi/paths/subscriptions@{id}@items@{itemId}.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+put:
+  x-badge: Experimental
+  x-products:
+    - Core
+  tags:
+    - Orders
+  summary: Update subscription order item
+  operationId: PutSubscriptionItem
+  x-sdk-operation-name: updateItem
+  description: Updates subscription order item. This is an experimental endpoint, can be changed or removed in the future.
```
Comment (noon-dawg) [inline openapi/paths/subscriptions@{id}@items@{itemId}.yaml:18]: nitpick: typo
```suggestion
  description: Updates a subscription order item. This is an experimental endpoint, can be changed or removed in the future.
```

File: openapi/paths/subscriptions@{id}@items@{itemId}.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    required: true
+    schema:
+      type: string
+put:
+  x-badge: Experimental
+  x-products:
+    - Core
+  tags:
+    - Orders
+  summary: Update subscription order item
```
Comment (noon-dawg) [inline openapi/paths/subscriptions@{id}@items@{itemId}.yaml:15]: suggestion: I think we should add 'a' to emphasize that this updates a singular order item.

We do try to keep summary descriptions short, and sometimes omit articles to do this, but I think it's useful here

```suggestion
  summary: Update a subscription order item
```

File: openapi/paths/subscriptions@{id}@items@{itemId}.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  requestBody:
+    content:
+      application/json:
+        schema:
+          $ref: ../components/requestBodies/SubscriptionItemUpdate.yaml
+    description: Update item request.
+    required: true
+  responses:
+    '201':
+      description: Order item update.
```
Comment (noon-dawg) [inline openapi/paths/subscriptions@{id}@items@{itemId}.yaml:28]: suggestion:
```suggestion
      description: Order item updated.
```

File: openapi/components/schemas/OrderItem.yaml [hunk 6]
```diff
@@ -18,6 +18,9 @@ properties:
   quantity:
     description: Number of product units in the specified plan.
     type: integer
+  quantityUsed:
```
Comment (adamaltman) [inline openapi/components/schemas/OrderItem.yaml:21]: I wonder to use the term "Used" or "Filled"?

```suggestion
  quantityFilled:
```

Comment (noon-dawg) [reply openapi/components/schemas/OrderItem.yaml:21]: Good point! 'Used' is ambiguous here. I think the term 'filled' more closely relates to the number of items in the order, and these items may not be 'used' yet.

We should also update [line 22](https://github.com/Rebilly/api-definitions/pull/1754/files#diff-2f5d627238c8032bc7c2dcf9595f83f6cad042a6c8119b492ba09d9642f3ccdbR22) to `Number of filled product units.`

File: openapi/components/schemas/OrderItem.yaml [hunk 7]
```diff
@@ -18,6 +18,9 @@ properties:
...
+    description: Number of used product units.
```
Comment (noon-dawg) [inline openapi/components/schemas/OrderItem.yaml:22]:
```suggestion
    description: Number of filled product units.
```

**Review by noon-dawg — DISMISSED** [review]

**Review by ashkarpetin — APPROVED** [review]

**Review by i-yasko — APPROVED** [review]

**Review by noon-dawg — APPROVED** [review]

**Pull request merged**